test/jdk/java/text/Format/ChoiceFormat/Bug8001209.java

```diff
@@ -21,70 +21,72 @@
  * questions.
  */
 
-/**
+/*
  * @test
  * @bug 8001209
  * @summary Confirm that the values set by setChoices() are not mutable.
+ * @run junit Bug8001209
  */
-import java.text.*;
 
-public class Bug8001209 {
+import java.text.ChoiceFormat;
+import java.text.ParsePosition;
+
+import org.junit.jupiter.api.Test;
 
-    public static void main(String[] args) throws Exception {
-        boolean err = false;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class Bug8001209 {
 
-        // Borrow an example in API doc
-        double[] limits = {1,2,3,4,5,6,7};
-        String[] dayOfWeekNames = {"Sun","Mon","Tue","Wed","Thu","Fri","Sat"};
+    /*
+     * Ensure that mutating the arrays passed to setChoices and setLimits as well
+     * as the arrays returned by getChoices and getLimits does not affect
+     * the internal representation of the original ChoiceFormat.
+     */
+    @Test
+    public void immutableChoicesAndFormatsTest() {
+        double[] limits = {1, 2, 3, 4, 5, 6, 7};
+        String[] dayOfWeekNames = {"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
         ChoiceFormat form = new ChoiceFormat(limits, dayOfWeekNames);
         ParsePosition status = new ParsePosition(0);
 
+        // Build the expected results of formatting with the original ChoiceFormat
         StringBuilder before = new StringBuilder();
         for (double i = 1.0; i <= 7.0; ++i) {
             status.setIndex(0);
             String s = form.format(i);
             before.append(" ");
             before.append(s);
-            before.append(form.parse(form.format(i),status));
+            before.append(form.parse(form.format(i), status));
         }
         String original = before.toString();
 
-        double[] newLimits = form.getLimits();
-        String[] newFormats = (String[])form.getFormats();
+        // Modify the array returned by getFormats() -> newFormats
+        String[] newFormats = (String[]) form.getFormats();
         newFormats[6] = "Doyoubi";
         StringBuilder after = new StringBuilder();
         for (double i = 1.0; i <= 7.0; ++i) {
             status.setIndex(0);
             String s = form.format(i);
             after.append(" ");
             after.append(s);
-            after.append(form.parse(form.format(i),status));
-        }
-        if (!original.equals(after.toString())) {
-            err = true;
-            System.err.println("  Expected:" + before
-                               + "\n  Got:     " + after);
+            after.append(form.parse(form.format(i), status));
         }
+        // Compare the expected results with the new formatted results
+        assertEquals(after.toString(), original,
+                "Mutating array returned from getter changed internals of ChoiceFormat");
 
+        // Modify the array passed to setFormats() -> dayOfWeekNames
```

Comment:
I think we should keep the `err` variable, and not throw an exception (and end the test) here. Otherwise, the test after this will not run.

Reply:
You're right, I should have actually split this larger test into two separate tests (one for the getter / one for the setter), then there would be no need for `err`.

I have updated the PR with the change I mentioned. Thank you for reviewing.